test/test_container.py

```diff
@@ -118,6 +120,11 @@ def logger():
         yield patched
 
 
+@pytest.fixture(autouse=True)
```

Comment:
i recently realised that these are silent by default, and wondering whether we want to change that. until recently django used to use a subclass to un-hiden deprecation warnings https://github.com/django/django/blob/1.9/django/utils/log.py#L55 (was changed in https://code.djangoproject.com/ticket/25999 to aid their LTS process which involves third parties conditionally importing old stuff which would trigger these warnings). i don't think we have that problem, so we might want to consider doing this